compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -130,9 +130,14 @@ class TypeComparer(initctx: Context) extends ConstraintHandling[AbsentContext] {
     }
   }
 
+  /** The current approximation state. See `ApproxState`. */
   private[this] var approx: ApproxState = FreshApprox
   protected def approxState: ApproxState = approx
 
+  /** The original left-hand type of the comparison. Gets reset
```

Comment:
Not sure "reset" is the correct word since the value is saved and restored when leaving the nested isSubType. In any case, I think we can avoid having this variable at all by nesting `isSubArgs` inside `isSubType` and rename the parameters of `isSubType` from `tp1` and `tp2` to `origTp1`, `origTp2`. Then `leftRoot` can just be replaced by `origTp1`.